spring-kafka/src/main/java/org/springframework/kafka/core/KafkaOperations.java

```diff
@@ -64,6 +64,16 @@
 	ListenableFuture<SendResult<K, V>> sendDefault(int partition, K key, V data);
 
 	/**
+	 * Send the data to the default topic with the provided key and partition.
+	 * @param partition the partition.
+	 * @param timestamp the timestamp of the record.
+	 * @param key the key.
+	 * @param data the data.
+	 * @return a Future for the {@link SendResult}.
+	 */
+	ListenableFuture<SendResult<K, V>> sendDefault(int partition, Long timestamp, K key, V data);
```

Comment:
Agree, since we are introducing the variant when you can use `null` for `partition` (via `Integer`) as you have wanted before with the `send(String topic, K key, V data)`, it should be fine to make this breaking change do not break minds in the future.

Your thoughts, @garyrussell ?